src/System.Private.ServiceModel/tools/test/Bridge/Bridge/ResourceController.cs

```diff
@@ -67,39 +82,62 @@ public HttpResponseMessage Put(HttpRequestMessage request)
                 Trace.WriteLine(String.Format("{0:T} - Exception executing PUT for resource {1}{2}:{3}",
                                                 DateTime.Now, resourceName, Environment.NewLine, exception.ToString()), 
                                 this.GetType().Name);
-                return request.CreateResponse(HttpStatusCode.InternalServerError, new resourceResponse
-                {
-                    id = correlationId,
-                    details = exception.Message
-                });
+                return request.CreateResponse(HttpStatusCode.InternalServerError, exception.ToString());
             }
         }
 
         /// <summary>
-        /// Gets resource information
+        /// Invoke the <c>Get</c> method of the <see cref="IResource"/>
+        /// matching the name specified in the content name/value pairs.
         /// </summary>
-        /// <param name="name">name of the resource</param>
-        public HttpResponseMessage Get(string name)
+        /// <param name="request">The incoming request containing the name/value pair content</param>
+        /// <returns>The response to return to the caller.</returns>
+        public HttpResponseMessage Get(HttpRequestMessage request)
         {
-            Trace.WriteLine(String.Format("{0:T} - Received GET request for resource {1}", 
-                                          DateTime.Now, String.IsNullOrWhiteSpace(name) ? "null" : name),
-                            this.GetType().Name);
-
-            if (string.IsNullOrWhiteSpace(name))
+            if (request.Content == null)
             {
-                Trace.WriteLine(String.Format("{0:T} - GET response is BadRequest due to missing name", DateTime.Now),
+                string badRequestMessage = "GET request did not contain content";
+                Trace.WriteLine(String.Format("{0:T} - {1}",
+                                              DateTime.Now,
+                                              badRequestMessage),
                                 this.GetType().Name);
-                return Request.CreateErrorResponse(HttpStatusCode.BadRequest, "name not specified.");
+                return request.CreateResponse(HttpStatusCode.BadRequest, badRequestMessage);
             }
 
-            try
+            string nameValuePairs = request.Content.ReadAsStringAsync().GetAwaiter().GetResult();
```

Comment:
Do GETs have bodies? 

I don't know that the HTTP spec forbids it, but I don't think I see it in common use (for example, if you use Fiddler, you'll see that you can't actually send anything in the body). Typically for a GET, all the name/value pairs are specified on the URI itself